Service/AlgoliaCredentialsManager.php

```diff
@@ -46,22 +46,24 @@ public function checkCredentialsWithSearchOnlyAPIKey(int $storeId = null): bool
      * @param string $class
      * @param int|null $storeId
      * @return void
-     * @throws NoSuchEntityException
      */
     public function displayErrorMessage(string $class, ?int $storeId = null): void
     {
-        $storeInfo = $storeId ? ' for store '. $this->storeManager->getStore($storeId)->getName() : '';
-        $errorMessage = '
+        try {
+            $storeInfo = $storeId ? ' for store '. $this->storeManager->getStore($storeId)->getName() : '';
+            $errorMessage = '
 ' . $class . ': Algolia credentials missing' . $storeInfo . '
   => You need to configure your credentials in Stores > Configuration > Algolia Search.';
 
-        if (php_sapi_name() === 'cli') {
-            $this->output->writeln($errorMessage);
+            if (php_sapi_name() === 'cli') {
+                $this->output->writeln($errorMessage);
 
-            return;
-        }
+                return;
+            }
 
-        $this->messageManager->addErrorMessage($errorMessage);
+            $this->messageManager->addErrorMessage($errorMessage);
+        } catch (NoSuchEntityException $exception) {
+        }
```

Comment:
I don't think we should bury this exception as unlikely as it may be? Maybe something like: 
```
$this->messageManager->addErrorMessage(__("Unable to locate store details: %1", $exception->getMessage()));
```
Or lift the `$errorMessage` scope up and concatenate the errors. (Because it might hide the original issue?)

Reply:
Makes sense, updating it.